dispatcher.go

```diff
@@ -149,42 +197,64 @@ func dispatchMessage(conn net.Conn, msg []byte) { //*gClient.Message) {
 	var peer *SctpConnections
 	p, ok := connections.Load(conn)
 	if !ok {
-		sctpLog.Infof("Notification for unknown connection")
+		logger.SctpLog.Infof("Notification for unknown connection")
 		return
 	} else {
 		peer = p.(*SctpConnections)
-		sctpLog.Warnf("Handle SCTP Notification[addr: %+v], peer %v ", conn.RemoteAddr(), peer)
+		logger.SctpLog.Warnf("Handle SCTP Notification[addr: %+v], peer %v ", conn.RemoteAddr(), peer)
 	}
 
 	if len(msg) == 0 {
-		sctpLog.Infof("send Gnb connection close message to AMF %v", peer)
-		t := gClient.Message{}
+		logger.SctpLog.Infof("send Gnb connection [%v] close message to all AMF Instances", peer)
+		t := gClient.SctplbMessage{}
 		t.VerboseMsg = "Bye From gNB Message !"
 		t.Msgtype = gClient.MsgType_GNB_DISC
 		t.SctplbId = os.Getenv("HOSTNAME")
-		t.GnbId = peer.address
+		t.GnbIpAddr = peer.address
 		t.Msg = msg
-		backend := backends[0]
-		backend.sctp = conn
-		if err := backend.stream.Send(&t); err != nil {
-			log.Fatalf("can not send %v", err)
+		if backends != nil && len(backends) > 0 {
+			var i int
+			for ; i < len(backends); i++ {
+				backend := backends[i]
+				if backend.state == true {
+					if err := backend.stream.Send(&t); err != nil {
+						logger.SctpLog.Errorln("can not send ", err)
+					}
+				}
+			}
+		} else {
+			logger.SctpLog.Errorln("No AMF Connections")
 		}
+		context.Sctplb_Self().DeleteRan(conn)
 		return
 	}
-	sctpLog.Println("Message received from remoteAddr ", conn.RemoteAddr().String())
-	t := gClient.Message{}
+	ran, _ := context.Sctplb_Self().RanFindByConn(conn)
+	if ran == nil {
+		ran = context.Sctplb_Self().NewRan(conn)
+	}
+	logger.SctpLog.Println("Message received from remoteAddr ", conn.RemoteAddr().String())
+	t := gClient.SctplbMessage{}
```

Comment:
the hello from GNB should be only for first message right ?

Reply:
No.. for every msg